scripts/submit_bb.py

```diff
@@ -36,129 +37,178 @@ def gen_command_template(params):
     return command_template_parameters, params.bb
 
 
-def main(args: argparse.Namespace, logger: Logger = get_basic_logger()):
-    params = utils.load_sim_params(os.path.join(args.rel_dir, "sim_params.yaml"))
-    ncores = platform_config[const.PLATFORM_CONFIG.BB_DEFAULT_NCORES.name]
+def main(
+    auto: bool = False,
+    machine: str = host,
+    ncores: int = platform_config[const.PLATFORM_CONFIG.BB_DEFAULT_NCORES.name],
+    rel_dir: str = ".",
+    retries: int = 0,
+    version: str = None,
+    write_directory: str = None,
+    logger: Logger = get_basic_logger(),
+):
+    rel_dir = Path(rel_dir).resolve()
+    try:
+        params = utils.load_sim_params(rel_dir / "sim_params.yaml")
+    except FileNotFoundError:
+        logger.error(f"Error: sim_params.yaml doesn't exist in {rel_dir}")
+        raise
+
+    params.sim_dir = Path(params.sim_dir).resolve()
```

Comment:
If we want to do this kind of transform I'd prefer it either stored as a local variable or as part of the load_sim_params. Modifying dictionaries (or namespaces) that are loaded from somewhere leads to potential confusion.

Reply:
ok - agreed.